backend/LexBoxApi/dev.Dockerfile

```diff
@@ -9,7 +9,7 @@ WORKDIR /src/backend
 # Copy the main source project files
 COPY */*.csproj *.sln ./
 # move them into the proper sub folders, based on the name of the project
-RUN for file in $(ls *.csproj); do dir=${file%.*} mkdir -p ${file%.*}/ && mv $file ${file%.*}/ && dotnet restore ${file%.*}/${file}; done
```

Comment:
why don't we just restore as a second step. It should pickup the sln and restore all projects. Then we don't have to do this weird echo into a shell script.

Reply:
The .sln file isn't in the `backend` directory, so it's not visible to the Dockerfile. We could use shenanigans to make it visible to the Docker build context, but I prefer to simply restore all the .csproj files. I tried running `dotnet restore */*.csproj` but `dotnet restore` wants only one .csproj file per line.

So I realized that we need to build up a list of projects to restore, and then restore them after the loop is done. I could have written that list into a text file and done `for item in $(cat file); do dotnet restore ${item}`, but making the list a shell script was far simpler. (There are non-trivial issues around filenames with spaces in that `for item in $(cat file)` syntax).

Reply:
then in that case let's just restore LexBoxApi. A simple `dotnet restore ./LexBoxApi/LexBoxApi.csproj` after the for loop is done is all we need. It'll trigger a restore of any projects it depends on (and none it doesn't).

Reply:
Okay. I thought it would be better to restore all projects so that if we add any later we don't have to remember to edit the dockerfile, but that's simple enough. For some reason, LexBoxApi.csproj doesn't depend on FixFwData.csproj so I'll need to add a restore step for that one too (it only matters for running Send/Receive, i.e. in the integration tests, but at least in the dev.Dockerfile we definitely want to be able to trigger a Send/Receive). But I'll do that rather than the bash script.

Reply:
actually we only need the FwData for running the tests, that's why the Api doesn't depend on it. And we're never going to run tests inside the backend container (against it yes, but not inside it) so that won't be an issue. Which is why any future projects that get added we don't want to restore either if LexBox doesn't depend on them.